### PR TITLE
Add GraphQLTelemetryOptions.RecordException configuration

### PR DESCRIPTION
**Pull request by gao-artur**

### CODE REVIEW

File: src/GraphQL/Telemetry/GraphQLTelemetryProvider.cs [hunk 1]
```diff
@@ -199,7 +192,29 @@ protected virtual Task SetResultTagsAsync(Activity activity, ExecutionOptions ex
...
+        if (_telemetryOptions.RecordException)
+        {
+            activity.RecordException(ex);
+        }
+
+        _telemetryOptions.EnrichWithException(activity, ex);
+        return Task.CompletedTask;
+    }
+
+    private async Task<bool> FilterAsync(ExecutionOptions options, Activity activity)
```
Comment (Shane32) [inline src/GraphQL/Telemetry/GraphQLTelemetryProvider.cs:217]:
```suggestion
    private async ValueTask<bool> FilterAsync(ExecutionOptions options, Activity activity)
```

File: src/GraphQL/Telemetry/GraphQLTelemetryProvider.cs [hunk 2]
```diff
@@ -199,7 +192,29 @@ protected virtual Task SetResultTagsAsync(Activity activity, ExecutionOptions ex
         return Task.CompletedTask;
     }
 
-    private bool Filter(ExecutionOptions options, Activity activity)
+    /// <summary>
+    /// Handle execution exception. The default implementation sets the <see cref="Activity"/> error status,
+    /// records exception event if <see cref="GraphQLTelemetryOptions.RecordException"/> is <see langword="true"/>
+    /// and executes the <see cref="GraphQLTelemetryOptions.EnrichWithException"/> callback.
+    /// </summary>
+    protected virtual Task OnException(Activity activity, Exception ex)
```
Comment (Shane32) [inline src/GraphQL/Telemetry/GraphQLTelemetryProvider.cs:200]:
```suggestion
    protected virtual Task OnExceptionAsync(Activity activity, Exception ex)
```

File: src/GraphQL/Telemetry/GraphQLTelemetryProvider.cs [hunk 3]
```diff
@@ -107,14 +107,7 @@ private async Task<ExecutionResult> ExecuteInternalAsync(ExecutionOptions option
...
             if (ex is not OperationCanceledException)
-            {
-#if NET6_0_OR_GREATER
-                activity.SetStatus(ActivityStatusCode.Error);
-#else
-                activity.SetTag("otel.status_code", "ERROR");
-#endif
-            }
-            _telemetryOptions.EnrichWithException(activity, ex);
+                await OnException(activity, ex).ConfigureAwait(false);
```
Comment (Shane32) [inline src/GraphQL/Telemetry/GraphQLTelemetryProvider.cs:110]:
```suggestion
                await OnExceptionAsync(activity, ex).ConfigureAwait(false);
```

File: src/GraphQL/Telemetry/GraphQLTelemetryProvider.cs [hunk 4]
```diff
@@ -212,12 +227,7 @@ private bool Filter(ExecutionOptions options, Activity activity)
         }
         catch (Exception ex)
         {
-#if NET6_0_OR_GREATER
-            activity.SetStatus(ActivityStatusCode.Error);
-#else
-            activity.SetTag("otel.status_code", "ERROR");
-#endif
-            _telemetryOptions.EnrichWithException(activity, ex);
+            await OnException(activity, ex).ConfigureAwait(false);
```
Comment (Shane32) [inline src/GraphQL/Telemetry/GraphQLTelemetryProvider.cs:230]:
```suggestion
            await OnExceptionAsync(activity, ex).ConfigureAwait(false);
```

File: src/GraphQL/Telemetry/GraphQLTelemetryOptions.cs [hunk 5]
```diff
@@ -23,6 +23,14 @@ public class GraphQLTelemetryOptions
     /// </summary>
     public Func<ExecutionOptions, string?> SanitizeDocument { get; set; } = options => options.Query;
 
+    /// <summary>
+    /// Gets or sets a value indicating whether the exception will be recorded as ActivityEvent.
+    /// </summary>
+    /// <remarks>
+    /// https://github.com/open-telemetry/semantic-conventions/blob/main/docs/exceptions/exceptions-spans.md.
+    /// </remarks>
+    public bool RecordException { get; set; }
```
Comment (Shane32) [inline src/GraphQL/Telemetry/GraphQLTelemetryOptions.cs:32]: Default true or false?

**Review by Shane32 — APPROVED** [review]

File: src/GraphQL/Telemetry/GraphQLTelemetryOptions.cs (hunk 5 above)
Comment (gao-artur) [reply src/GraphQL/Telemetry/GraphQLTelemetryOptions.cs:32]: False

**Pull request merged**